.github/actions/check-run/action.yml

```diff
@@ -0,0 +1,59 @@
+name: Check Run
+description: Creates markdown report via the Checks API
+# https://docs.github.com/en/rest/reference/checks#runs
+
+inputs:
+  repository:
+    description: The GitHub repository {owner}/{repo}
+    required: false
+    default: ${{ github.repository }}
+  token:
+    description: Github token to use to publish the check.
+    required: false
+    default: ${{ github.token }}
+  name:
+    description:  The name of the check.
+    required: true
+  sha:
+    description: The SHA of the commit.
+    required: false
+    default: ${{ github.ref }}
+  conclusion:
+    description: The final conclusion of the check. Can be one of action_required, cancelled, failure, neutral, success, skipped, stale, or timed_out.
+    required: false
+    default: ${{ job.status }}
+  title:
+    description: The title of the check run.
+    required: true
+  markdown:
+    description: Path to the Markdown summary of the check run.
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Setup jq
+      run: apt-get install jo -y
+      shell: bash
+
+    - name: Publish report summary
```

Comment:
Does this create the report in a nice looking manner from the data.json?

Reply:
Correct, the report is contained in `summary` and `data.json` contains other information about the check